src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,79 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class Parser {
+    private boolean bye;
+
+    public Parser() {
+        this.bye = false;
+    }
+
+    public boolean isBye() {
+        return this.bye;
+    }
+
+    public void parse(String input, TaskList tasks) throws DukeException {
+        if (input.equals("bye")) {
+            this.bye = true;
+        } else if (input.equals("list")) {
+            tasks.showList();
+        } else if (input.startsWith("mark")) {
+            if (input.length() < 6) {
```

Comment:
We should try to minimize hardcoding numbers without meaning into our code. Check out the `replace` function, it might be beneficial to replace the string "mark" with an empty string so we can check for the task given after mark.